### ISSUE TITLE
Add Prompt testing using a LLM testing suite

### ISSUE THREAD

**Issue by vmesel** [talkdai/dialog#174]
We need to enable our users to prevent having regressions on their prompt without noticing it clearly. In order to achieve this, we must implement a way for our users to run a test suite on their own test cases with a user set similarity score. This must be simply set and must be extensible to use beyond postgresql.

**Comment by avelino** [comment]
https://github.com/confident-ai/deepeval?tab=readme-ov-file#writing-your-first-test-case

this week @lgabs shared this project with me, apparently we were able to use it to test our prompts

**Comment by vmesel** [comment]
Yes, I saw the README on ragtalks and forgot to attach the link, thats the
library I was looking for.

**Comment by lgabs** [comment]
Yeah, I couldn't study too much of llm evals, but I did check that the way the community seams to evaluate llms applications is using several standard metrics such that another llm evaluates the llm apppication outputs against expected results (this evaluator could even be a free local llm, for a task is much simpler).

I saved this [Andre Ng short course ](https://learn.deeplearning.ai/courses/automated-testing-llmops?_gl=1*12438vp*_ga*MTEwODUyNjk1NS4xNzEyNDk1ODIw*_ga_PZF1GBS1R1*MTcxMjQ5NTgyMC4yLjEuMTcxMjQ5NTg5Ny41OS4wLjA.)to see soon, maybe it'll help   🤘

**Comment by lgabs** [comment]
Also, I think it would be a good idea for us to use the same common dataset for local dev, and also for tests that depend on dataset (generate embeddings or even these llm evals), one idea is to download some one from hugging face like this [wiki_qa](https://huggingface.co/datasets/wiki_qa. What do you think? This would a new issue, of course.

These llm evals change a lot depending on the domain, so i think we could just make good documentation about how to add these tests without trying to write generic cases for all datasets.

**Comment by vmesel** [comment]
So I was thinking on making it available for the developer to create new
test cases, not necessarily our software writing generic test cases.

Think of having just a single test case, instead of having to implement it
by yourself inside dialog, you can just write a new toml file that could
enable this feature.

On using other LLMs, I’m not aware on what should be implemented in those
cases, going to watch the video here to research more about this.
